client/src/views/devices/ImportRecoveryDevicePage.vue

```diff
@@ -0,0 +1,278 @@
+      </ion-card-title>
+      <ms-informative-text>
+        {{ $t('ImportRecoveryDevicePage.subtitles.passwordModified') }}
+      </ms-informative-text>
+      <ion-button @click="goToHome()">
+        {{ $t('ImportRecoveryDevicePage.actions.goBackToLogin') }}
+      </ion-button>
+    </ion-card>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { asyncComputed } from '@/common/asyncComputed';
+import { Validity, secretKeyValidator } from '@/common/validators';
+import { MsChoosePasswordInput, MsInformativeText, MsInput, MsReportText, MsReportTheme } from '@/components/core';
+import { AvailableDevice, DeviceInfo, RecoveryImportErrorTag, SecretKey, deleteDevice, importRecoveryDevice, saveDevice } from '@/parsec';
+import { Notification, NotificationKey, NotificationLevel, NotificationManager } from '@/services/notificationManager';
+import { IonButton, IonCard, IonCardContent, IonCardTitle, IonIcon, IonText } from '@ionic/vue';
+import { checkmarkCircle, document, key } from 'ionicons/icons';
+import { Ref, defineEmits, defineProps, inject, onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+const { t } = useI18n();
+
+enum ImportDevicePageState {
+  Start = 'start',
+  Password = 'password',
+  Done = 'done',
+}
+
+const state = ref(ImportDevicePageState.Start);
+const choosePasswordInput: Ref<typeof MsChoosePasswordInput | null> = ref(null);
+const hiddenInput = ref();
+const secretKey: Ref<SecretKey> = ref('');
+// cspell:disable-next-line
+const secretKeyPlaceholder = 'FH3H-N3DW-RUOO-A6Q7-...';
+const changeButtonIsEnabled = asyncComputed(async (): Promise<boolean> => {
+  return choosePasswordInput.value && (await choosePasswordInput.value.areFieldsCorrect());
+});
+const recoveryFile: Ref<File | null> = ref(null);
+const newDeviceInfo: Ref<DeviceInfo | null> = ref(null);
+const isSecretKeyValid = ref(false);
+const notificationManager: NotificationManager = inject(NotificationKey)!;
+
+const emits = defineEmits<{
+  (e: 'finished'): void;
+}>();
+
+const props = defineProps<{
+  device: AvailableDevice;
+}>();
+
+onMounted(() => {
+  hiddenInput.value.addEventListener('change', onInputChange);
+});
+
+async function onInputChange(_event: Event): Promise<void> {
+  if (hiddenInput.value.files.length === 1) {
+    recoveryFile.value = hiddenInput.value.files[0];
+  }
+}
+
+async function importButtonClick(): Promise<void> {
+  hiddenInput.value.click();
+}
+
+async function checkSecretKeyValidity(): Promise<void> {
+  isSecretKeyValid.value = (await secretKeyValidator(secretKey.value)) === Validity.Valid;
+}
+
+async function goToPasswordChange(): Promise<void> {
+  if (recoveryFile.value) {
+    const result = await importRecoveryDevice(props.device.deviceLabel, recoveryFile.value, secretKey.value);
+    if (result.ok) {
+      newDeviceInfo.value = result.value;
+      state.value = ImportDevicePageState.Password;
+    } else {
+      if (result.error.tag === RecoveryImportErrorTag.KeyError) {
```

Comment:
Don't forget to handle the `internal` tag. Probably better to use a `switch()` here and refactor a little.